alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/settings/SupportedApiSettings.kt

```diff
@@ -0,0 +1,66 @@
+package com.amazon.opendistroforelasticsearch.alerting.settings
+
+import com.amazon.opendistroforelasticsearch.alerting.core.model.LocalUriInput
+
+/**
+ * A class that supports storing a unique set of API paths that can be accessed by general users.
+ */
+class SupportedApiSettings {
+    companion object {
+        const val CLUSTER_HEALTH_PATH = "/_cluster/health"
+        const val CLUSTER_STATS_PATH = "/_cluster/stats"
+
+        /**
+         * Each String represents the path to call an API.
+         * NOTE: Paths should conform to the following pattern:
+         * "/_cluster/health"
+         *
+         * Each Set<String> represents the supported JSON payload for the respective API.
+         */
+        private var supportedApiList = HashMap<String, Map<String, Set<String>>>()
+
+        /**
+         * Set to TRUE to enable the supportedApiList check. Set to FALSE to disable.
+         */
+        // TODO: Currently set to TRUE for testing purposes.
+        //  Should likely be set to FALSE by default.
+        private var supportedApiListEnabled = true
```

Comment:
just curious: why do we need this?

Reply:
This is essentially an on/off switch that I implemented at the beginning of the project. If a user did not want to check the API called by the monitor against the allow list, this value could be set to FALSE. That would allow users to create monitors for any API that this feature support. At the moment, it doesn't serve much of a purpose; but it could be more valuable as more API become supported.

We could remove this value if we feel it's extraneous.

Reply:
If I understand this correctly this is something intended to be specified by the user and if this is set to false then they cannot use the LocalUrlInput feature.

I am not sure I understand how this is configurable by the end user at the moment. Do you think this can be defined as a cluster setting so user can turn it on off as it seems fit for them? or if this is not configurable at the moment suggest going with removing this as it serves no purpose and in future when needed to add the customization we can introduce the cluster setting

Reply:
If configured to false, users could still use the LocalUriInput feature; but the allow list wouldn't be checked for the requested API before creating and executing the monitor. At the moment, the allow list exposes both of the supported API; but as support for more API is added, a cluster owner could essentially configure this setting to false to allow access to any supported API, regardless of whether they are on the allow list.

Your comment about configuring via settings makes sense though. This will be easy enough to reimplement if there's a need in the future. 

Removed this value.